codegen/src/main/java/software/amazon/awssdk/codegen/poet/rules/EndpointResolverInterceptorSpec.java

```diff
@@ -484,6 +541,33 @@ private MethodSpec setContextParamsMethod(OperationModel opModel) {
         return b.build();
     }
 
+    private MethodSpec setOperationContextParamsMethod(OperationModel opModel) {
+        String requestClassName = model.getNamingStrategy().getRequestClassName(opModel.getOperationName());
+        ClassName requestClass = poetExtension.getModelClass(requestClassName);
+
+        MethodSpec.Builder b = MethodSpec.methodBuilder("setOperationContextParams")
+                                         .addModifiers(Modifier.PRIVATE, Modifier.STATIC)
+                                         .addParameter(paramsBuilderClass(), "params")
+                                         .addParameter(requestClass, "request")
+                                         .returns(void.class);
+
+        opModel.getOperationContextParams().forEach((key, value) -> {
+            String setterName = endpointRulesSpecUtils.paramMethodName(key);
+
+            //TODO: JMESPathRunTime PR for OperationContextParam will change this to extract List based on JMESPath expression
+            // instead of the new ArrayList().
+            if (Objects.requireNonNull(value.getValue().asToken()) == JsonToken.VALUE_STRING) {
+                b.addComment("TODO: Add JMESPathRuntime for $L", ((JrsString) value.getValue()).getValue());
+            } else {
+                throw new RuntimeException("Don't know how to set parameter of type " + value.getValue().asToken());
```

Comment:
I feel like people could make this mistake. Can we be more useful in the error message? Something like... "Invalid operation context parameter path for {operation-name}. Expected VALUE_STRING, but got {actual-value}".